[NEW] Option to enable/disable auto away and configure timer

armand1m opened the pull request:
<!-- INSTRUCTION: Keep the line below to notify all core developers about this new PR -->
@RocketChat/core 

<!-- INSTRUCTION: Inform the issue number that this 
 
PR closes, or remove the line below -->
Closes #736 

<!-- INSTRUCTION: Tell us more about your PR with screen shots if you can -->
This PR adds two new options to the Account Preferences, where the user can:

 - Enable Auto Away
 - Configure a default Idle Time Limit (only works when auto away is enabled, of course)

This PR also fixes a bug I've found in the `checked` template helper of the account settings package. When values where persisted as false values, but their default ones where `true` values, the `checked` helper would return always the default value since the way it was accessing the nested properties and verifying if they exists where the same as searching for its value.

![screenshot_27](https://user-images.githubusercontent.com/6154901/30006386-c1b8d9dc-90cd-11e7-9ea5-2a07c574f588.png)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=8029) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/RocketChat/Rocket.Chat?pullRequest=8029) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RocketChat/Rocket.Chat?pullRequest=8029) it.</sub>

rodrigok commented on packages/rocketchat-i18n/i18n/en.i18n.json:
```diff
@@ -568,6 +568,8 @@
   "Enable_two-factor_authentication": "Enable two-factor authentication",
   "Enabled": "Enabled",
+  "Enable_Auto_Away": "Enable Auto Away",
+  "Enable_Svg_Favicon": "Enable SVG favicon",
```
Is this string addition correct?

armand1m replied:
I believe that the "Enable Svg Favicon" came from a merge made to resolve some conflicts. I may need to look at this PR again now :/

rodrigok commented on tests/pageobjects/side-nav.page.js:
```diff
@@ -39,6 +39,10 @@ class SideNav extends Page {
 
+	get accountStatusBullet() { return browser.element('.sidebar__account-status-bullet'); }
+	get accountStatusBulletOnline() { return browser.element('.sidebar__account-status-bullet--online'); }
+	get accountStatusBulletAway() { return browser.element('.sidebar__account-status-bullet--away'); }
```
Why did you add this?

armand1m replied:
Hi, this was meant to be used in e2e tests, but I actually saw that there already were some tests for this functionality. I should have not commited this, I'm removing it.

armand1m commented:
Hey, sorry, I've been busy with work. Today I'll review this PR and update it accordingly, as well as fix the broken tests

rodrigok commented:
@armand1m It's awesome to know that you are working on it again. Let me know if you can fix that things soon, I'm merging PRs to ship a new release candidate tomorrow and I'd love to include your PR.

armand1m commented:
@rodrigok nice, I'll give it some attention now so we can close this issue. Expect to have this PR prepared today.

armand1m commented:
@rodrigok Got the tests passing 😄 

I've cleaned the dead code that I've left in the first version. I've also fixed some other tests mistakes.
I've implemented internationalization of this feature in 4 languages:

- English (default)
![captura de tela 2017-12-04 as 23 04 00](https://user-images.githubusercontent.com/6154901/33587513-9cb2f354-d955-11e7-8be3-d12ae77f07e8.png)

- Portuguese
![captura de tela 2017-12-04 as 23 04 12](https://user-images.githubusercontent.com/6154901/33587493-8792db9c-d955-11e7-9707-9b6264215859.png)

- Brazilian Portuguese
![captura de tela 2017-12-05 as 00 42 34](https://user-images.githubusercontent.com/6154901/33587474-71d35570-d955-11e7-899c-cdff9006f988.png)

- Dutch (Nederlands)
![captura de tela 2017-12-05 as 00 42 23](https://user-images.githubusercontent.com/6154901/33587476-7747a8ee-d955-11e7-9091-9d067185eaa4.png)

Here is a GIF demonstrating the feature.

![enableautoaway](https://user-images.githubusercontent.com/6154901/33587771-bef6c020-d956-11e7-97fa-f2ab507be75f.gif)

armand1m commented:
Hope to see it released tomorrow :smile:

rodrigok commented on packages/rocketchat-ui-account/client/accountPreferences.js:
```diff
@@ -152,19 +179,43 @@ Template.accountPreferences.onCreated(function() {
 		data.notificationsSoundVolume = parseInt($('#notificationsSoundVolume').val());
-
 		data.roomCounterSidebar = $('#roomCounterSidebar').find('input:checked').val();
+		data.messageCounterSidebar = $('#messageCounterSidebar').find('input:checked').val();
```
What is that item? I think it's old code, was replaced by `roomCounterSidebar`

armand1m replied:
Must have been placed in the last merge conflict I needed to solve. Sorry about that.

rodrigok replied:
No problem, it was fixed

armand1m commented:
Ok, guess it can be merged 😄
